### ISSUE TITLE
Queries from the Node.js BigQuery client v2.0.0 and higher are not returning results

### ISSUE THREAD

**Issue by d-torres** [googleapis/nodejs-bigquery#463]
Running a query in a Node.js 8 environment using the BigQuery Client Library, v3.0.0 results in an empty response returned to the caller.

Reproduction steps and observed behavior: 

Sample query: "with foo as (select 'Travis' as name) select count(*) as total from foo"

Expected behavior:  The query() method should return COUNT = 1 after running the aggregate query from above.  

Actual behavior: It's currently returning empty for bq client 2.0.0 and above.

**Comment by sduskis** [comment]
@callmehiphop, can you please take a look at this?

**Comment by callmehiphop** [comment]
@d-torres I just tested this against master and got the following result

```json
[
  {
    "total": 1
  }
]
```

**Comment by stephenplusplus** [comment]
I'm not sure how, but the exact version of Node 8 might matter, since we only support 8.10 and up: https://github.com/googleapis/nodejs-bigquery/blob/d64d892d7449c6b0df601f43d2afef3221dd5589/package.json#L8.

**Comment by JustinBeckwith** [comment]
The only reason that was set to `8.10` was because of HTTP/2 in the APIs that used `grpc-js`.  Shouldn't matter here I don't think, but good to know regardless.

**Comment by cmath** [comment]
Same thing happened to us. All of a sudden Select queries stopped working when running from cloud functions. Inserts are fine. When testing locally everything works as it should...

**Comment by d-torres** [comment]
@callmehiphop - Could we take a look at this from the context of a cloud function to see if we're able to reproduce?

**Comment by callmehiphop** [comment]
Confirmed not to be working on GCF, using the `query` method I started hitting a ` Maximum call stack size exceeded` error when requesting a single page of results. For a temporary workaround users can create a query job like so

```js
bigquery.createQueryJob(`SELECT * FROM foo`, (err, job) => {
  if (err) // error handling omitted
  job.getQueryResults((err, rows) => {
    if (err) // error handling omitted
  });
});
```

**Comment by callmehiphop** [comment]
Looks like the culprit is [this line](https://github.com/googleapis/nodejs-bigquery/blob/master/src/index.ts#L1592), it causes projectify to be called recursively quite a bit. For a fix I want to suggest not caching a job instance, but rather the job ID which we can use to grab a new job instance.

**Comment by melissammbeff** [comment]
I'm using @google-cloud/bigquery": "^3.0.0" and I'm having exactly this problem calling createQueryStream function. What should I do?

**Comment by callmehiphop** [comment]
@melissammbeff are you able to upgrade to the latest version (4.4.0)?

**Comment by melissammbeff** [comment]
Hi @callmehiphop, so, unfortunately not so far. I understood that this problem was solved on 2.0.6, so, why I'm still having the same problem on 3.0.0?

**Comment by callmehiphop** [comment]
@melissammbeff I'm not sure where `2.0.6` came from, but the PR that resolved this issue (#465) was not released until `4.1.1`.
